### PR TITLE
Display Cloud banner on self-hosted instances

### LINKED ISSUE #563: Display Cloud banner by default on self-host + revamp copy
**Expected behavior**
- The Cloud banner is displayed by default when running a **self-hosted** instance (see attached screen)
    - It should not be displayed if the mini-dashboard is opened for a **cloud instance**; a query parameter exists for that `cloud_banner=false`
- The banner can be closed and not displayed again
- Banner's copy changes
    - Banner title: "Scale up with Meilisearch Cloud 🚀" 
    - Banner text: "Faster, smarter search—no maintenance needed. [Start free] with no commitment." (use current link)

**Current behavior**
- The banner does not show up by default when opening the mini-dashboard on a **self-hosted** instance

**Screenshots**
Screenshot 2024-11-21 at 11 31 04

### PR DESCRIPTION
**Pull request by Strift**
# Pull Request

## Related issue
Fixes #563 

## What does this PR do?
- Display Cloud banner by default
- Hide Cloud banner only if query parameter has `cloud_banner=false`
- Update copy

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ import Modal from 'components/Modal'
 import NoMeilisearchRunning from 'components/NoMeilisearchRunning'
 import ApiKeyAwarenessBanner from 'components/ApiKeyAwarenessBanner'
 import getIndexesListWithStats from 'utils/getIndexesListWithStats'
-import shouldDisplayCloudBanner from 'utils/shouldDisplayCloudBanner'
+import isBannerEnabled from 'utils/isBannerEnabled'
```
Comment (mdubus) [inline src/App.js:19]: Can we rename it to have `cloud` in its name? 
If tomorrow we want to display an other banner too, it could be confusing :)

**Comment by mdubus** [comment]
It's not related to this PR, but scrolling makes the banner disappear 🤔 
I think that's not a classical behaviour for a banner and this is something we might want to fix. 
@Strift feel free to fix it here if that's an easy one, or to open an issue instead (could be a good first issue for contributors too 👍 ). 

https://github.com/user-attachments/assets/000d7fe4-92e8-4e09-a366-e36080c1428d


Otherwise, that's looking good to me 🙌

File: src/App.js (hunk 1 above)
Comment (Strift) [reply src/App.js]: Thanks for the feedback!

**Comment by mdubus** [comment]
> hey @mdubus, thanks for the review. As request I:
> 
> * made the banner sticky again (sorry for the oversight)
> * renamed the util file to include `Cloud` to make things clearer

Thank you @Strift  🙏 
But now if I close it, it breaks the UI:
https://github.com/user-attachments/assets/c91bf953-77d7-4c77-b5d7-b30d17a42ac5

Feel free to revert the change and open an issue instead to not spend too much time on it ;)

**Comment by curquiza** [comment]
> Feel free to revert the change and open an issue instead to not spend too much time on it ;)

I confirm, I think it's better so that we ensure to integrate this fix in v1.12 of Meilisearch 😊

**Comment by Strift** [comment]
Sorry for the oversight again. I tried to fix #565 to keep gaining familiarity with the code base.

If this is not working, I'll revert the changes so we can publish ASAP.

**Review by mdubus — APPROVED** [review]
LGTM ✨🦕 🌻
And thank you @Strift for taking care of those changes 🙏 👏

**Pull request merged**